applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -28,22 +28,34 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_USB_STATE_LOG_LEVEL);
 #include "config_event.h"
 #include "power_event.h"
 
+#if CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION
+  #include "usb_state_def.h"
+#else
+  #if (CONFIG_DESKTOP_HID_STATE_ENABLE) && (CONFIG_USB_HID_DEVICE_COUNT > 1)
+    #error USB selective HID subscription must be enabled.
+  #endif
+#endif /* CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION */
+
 #define REPORT_TYPE_INPUT	0x01
 #define REPORT_TYPE_OUTPUT	0x02
 #define REPORT_TYPE_FEATURE	0x03
 
 
 #ifndef CONFIG_USB_HID_PROTOCOL_CODE
-#define CONFIG_USB_HID_PROTOCOL_CODE -1
+  #define CONFIG_USB_HID_PROTOCOL_CODE -1
 #endif
 
+#if DT_PROP(DT_NODELABEL(usbd), num_in_endpoints) < (CONFIG_USB_HID_DEVICE_COUNT + 1)
+  #error Too few USB IN Endpoints enabled. Modify dts.overlay file.
+#endif
 
 struct usb_hid_device {
 	const struct device *dev;
 	uint8_t hid_protocol;
 	uint8_t sent_report_id;
 	bool report_enabled[REPORT_ID_COUNT];
 	bool enabled;
+	uint32_t report_bm;
```

Comment:
this should go right behind dev to match its alignment

Reply:
done